### PR TITLE
[EngSys] Package Updates

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to bump the version of `Azure.Core` referenced to match the latest stable release.  The version of `Microsoft.Azure.Amqp` has also been bumped to the latest stable release to
absorb several important bug fixes.

### CODE REVIEW

**Review by annelo-msft — APPROVED** [review]
LGTM.  Did you check whether any clients are using project references to Azure.Core?  We used to update those when we bumped the version, but IDK if we stopped doing that once engsys started blocking release of packages that still referenced the project and just let individual client release managers handle their own client project files.

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,8 @@
 
 ### Other Changes
 
+- Updated the `Microsoft.Azure.Amqp` dependency to 2.6.4, which includes several fixes.  One notable fix addresses an obscure race condition when a cancellation token is signaled while service operations are being invoked concurrently which caused those operations to hang.
```
Comment (JoshLove-msft) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/CHANGELOG.md:13]: Same comment here about this not being a direct dependency.

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,8 @@
 
 ### Other Changes
 
+- Updated the `Microsoft.Azure.Amqp` dependency to 2.6.4, which includes several fixes.  One notable fix addresses an obscure race condition when a cancellation token is signaled while service operations are being invoked concurrently which caused those operations to hang.  Another notable fix is for an obscure race condition that occurred when attempting to complete a message which caused the operation to hang.
```
[the remark this answers is not in the document]
Comment (jsquire) [reply sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/CHANGELOG.md:13]: Reworded to make clear that its an indirect dependency pulled in via update to the Service Bus package.   The fix is important enough that I'd like it visible in the change log.   Thoughts on the new phrasing?

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/CHANGELOG.md [hunk 3]
```diff
@@ -10,6 +10,8 @@
 
 ### Other Changes
 
+- Updated the `Microsoft.Azure.Amqp` dependency to 2.6.4, which includes several fixes.  One notable fix addresses an obscure race condition when a cancellation token is signaled while service operations are being invoked concurrently which caused those operations to hang.
```
Comment (m-redding) [inline sdk/eventhub/Azure.Messaging.EventHubs.Processor/CHANGELOG.md]: nit - was the version updated from 2.6.4 to 2.6.5 or am I reading that wrong?
```suggestion
- Updated the `Microsoft.Azure.Amqp` dependency to 2.6.5, which includes several fixes.  One notable fix addresses an obscure race condition when a cancellation token is signaled while service operations are being invoked concurrently which caused those operations to hang.
```

Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs.Processor/CHANGELOG.md]: You commented just before I pushed the fix for that.   😄

**Review by JoshLove-msft — APPROVED** [review]

**Pull request merged**